pkg/tfbridge/provider.go

```diff
@@ -1433,6 +1444,73 @@ func (p *Provider) Read(ctx context.Context, req *pulumirpc.ReadRequest) (*pulum
 	return &pulumirpc.ReadResponse{}, nil
 }
 
+// processImportValidationErrors runs `Validate` and then processes any resulting validation errors.
+// This only needs to run during an `Import` operation because the values read from the cloud
+// during import will be turned into the inputs in the generated code. Sometimes the cloud returns invalid
+// input values which will cause the next `up` to fail.
+//
+// To try and avoid this failure case, we run the validate functions for the generated inputs and
+// if any fail (and they match our other cases) then we remove those inputs so that the generated
+// program code will not contain the invalid inputs
+//
+// NOTE: we want to (for now at least) scope this as narrowly as possible because we do not want to
+// create a scenario where the resulting inputs create a different error
+// (e.g. removing an invalid required property). In these cases we will just allow the invalid configuration
+// through and the user will have to decide what to do.
+func (p *Provider) processImportValidationErrors(
+	ctx context.Context,
+	urn resource.URN,
+	tfName string,
+	inputs resource.PropertyMap,
```

Comment:
Maybe just https://github.com/pulumi/pulumi-terraform-bridge/blob/t0yv0%2Fautofill/unstable/propertyvalue/propertyvalue.go#L94 RemoveSecrets over the value here, if that's where secrets are coming from? This code should be obviously safe to ignore them.

Reply:
It looks like this works too, and I like it much better!

Reply:
Done! Let me know what you think.

Reply:
That's perfect, ty.